tests/core/api/test_userportalsync_create.py

```diff
@@ -0,0 +1,236 @@
+from fastapi.testclient import TestClient
+from sqlalchemy.orm import sessionmaker
+from pyispyb.app.extensions.database.middleware import Database
+from pyispyb.app.main import app
+from pyispyb.config import settings
+from pyispyb.app.extensions.database.session import engine
+from pyispyb.core.modules.proposals import get_proposals
+from pyispyb.core.modules.persons import get_persons
+from pyispyb.core.modules.laboratories import get_laboratories
+from pyispyb.core.modules.proteins import get_proteins
+from pyispyb.core.modules.sessions import get_sessions
+from pyispyb.core.modules.labcontacts import get_labcontacts
+from tests.core.api.utils.permissions import mock_permissions
+from tests.core.api.data.userportalsync_create import (
+    test_data_proposal_userportalsync_create,
+)
+
+
+TestingSessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+session = TestingSessionLocal()
+db = Database()
+db.set_session(session)
+
+
+client = TestClient(app)
+
+
+def test_call_sync_proposal_create():
```

Comment:
There are some helpers you can use to write tests:

- AuthClient can handle authentication for you: [authclient.py](https://github.com/ispyb/py-ispyb/blob/64f4b5b0aa568f3f9991a4651837f19e7a40c21e/tests/authclient.py)
- And if your test is a simple HTTP call to the API and then check code or return value, then you can use the helpers in [apitest.py](https://github.com/ispyb/py-ispyb/blob/64f4b5b0aa568f3f9991a4651837f19e7a40c21e/tests/core/api/utils/apitest.py) - see [example of usage (code)](https://github.com/ispyb/py-ispyb/blob/64f4b5b0aa568f3f9991a4651837f19e7a40c21e/tests/core/api/test_proposals.py) and [example of usage (data)](https://github.com/ispyb/py-ispyb/blob/64f4b5b0aa568f3f9991a4651837f19e7a40c21e/tests/core/api/data/proposals.py)

Reply:
Ok, I'll check that out and change accordingly.

Reply:
I just realized apitest.py only allows to use GET for now and you're doing a POST in this test, so it will not work as it currently is. But we could add a parameter in the input specification to determine what method should be used, and add a data payload input.

Reply:
`def test_call_sync_proposal_create(client):`

Reply:
Fixed with https://github.com/ispyb/py-ispyb/pull/210/commits/cf68b9ee40085dbad7913a6a9a08b22fa691bf3b